tests/test_table_sql.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import absolute_import, division, print_function
+import unittest
+
+import numpy as np
+import os
+from sqlalchemy import create_engine
+import lsst.utils.tests
+import afw_table as afw_table
+
+from lsst.daf.io.table.sql import get_schema, to_sql, read_sql
+
+
+class TableIoTestCase(lsst.utils.tests.TestCase):
+
+    def test_write(self):
+        """Test that a Schema can be read from a FITS file
+
+        Per DM-8211.
+        """
+        schema = afw_table.Schema()
+        aa = schema.addField("a", type=np.int64, doc="a")
+        bb = schema.addField("b", type=np.float64, doc="b")
+        cat = afw_table.BaseCatalog(schema)
+        row = cat.addNew()
+        row.set(aa, 12345)
+        row.set(bb, 1.2345)
+        row = cat.addNew()
+        row.set(aa, 4321)
+        row.set(bb, 4.123)
+
+        # Should use sqlite as default
+        get_schema(cat, "testname")
+        engine = create_engine("mysql://localhost")
+
+        # Should use MySQL dialect
+        get_schema(cat, "testname", con=engine)
+        location = os.path.dirname(__file__) + "/test.db"
+        engine = create_engine("sqlite:///" + location)
+
+        to_sql(cat, "testname", engine)
+        rows = engine.execute("select a, b from testname")
+        for sql_row, afw_row in zip(rows, cat):
+            self.assertEquals(sql_row["a"], afw_row["a"])
+            self.assertEquals(sql_row["b"], afw_row["b"])
+
+    def test_read(self):
+        schema = afw_table.Schema()
+        aa = schema.addField("a", type=np.int64, doc="a")
+        bb = schema.addField("b", type=np.float64, doc="b")
+        cat_expcted = afw_table.BaseCatalog(schema)
+        row = cat_expcted.addNew()
+        row.set(aa, 12345)
+        row.set(bb, 1.2345)
+        row = cat_expcted.addNew()
+        row.set(aa, 4321)
+        row.set(bb, 4.123)
+
+        location = os.path.dirname(__file__) + "/test.db"
```

Comment:
because this depends on a file that was written by the test_write test case, you probably should just make this all one test case called something like test_write_read.
you also need to add a tearDown function to delete the file that was written.